test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1939,7 +1940,8 @@ public synchronized String startNode(Settings settings) {
         }
         final List<NodeAndClient> nodes = new ArrayList<>();
         final int prevMasterCount = getMasterNodesCount();
-        int bootstrapMasterNodeIndex = prevMasterCount == 0 && autoManageMinMasterNodes && newMasterCount > 0
+        int bootstrapMasterNodeIndex = prevMasterCount == 0 && autoManageMinMasterNodes && newMasterCount > 0 && Arrays.stream(settings)
+            .allMatch(s -> Node.NODE_MASTER_SETTING.get(s) == false || TestZenDiscovery.USE_ZEN2.get(s) == true)
```

Comment:
Should we run Zen1IT with `autoManageMinMasterNodes` disabled? It's doing odd stuff, e.g. when restarting a node in a 2 node cluster, it changes min_master_nodes to 1 and triggers the Zen2 node to bootstrap all by itself, although it probably shouldn't?

Reply:
I think it does the right thing with `min_master_nodes`. I didn't think it bootstrapped the Zen2 node since there's already a master node present.

Reply:
Ok, I think 053d74684490388e225a01b3bc114ac02f9bc680 is sufficient.